### ISSUE TITLE
Support for TYPEOF and subqueries

### ISSUE THREAD

**Issue by marcellcsiszar** [jongpie/NebulaQueryAndSearch#27]
@jongpie Thanks for this useful querybuilder. I started utilizing it in my current project's soqlhandlers and I see at the moment 2 areas for improvement: supporting subqueries and typeof in case of polymorf relationships. What is your opinion?

**Comment by jongpie** [comment]
@marcellcsiszar, glad to hear that the project is useful! I agree that both subqueries and typeof. I had support for subqueries in an older version of the library, but it was removed when I recently rewrote the library. I'll re-add this feature soon.

For typeof support, I'll need to do some research to figure out the best way to handle it, but I think it'd be great to have this feature. 

I'll update this issue with my progress on both items. Thanks for the input!

**Comment by jongpie** [comment]
@marcellcsiszar I've just released [version 3 of this library](https://github.com/jongpie/NebulaQueryAndSearch/releases/tag/v3.0.0-beta) and I have started 2 new pull requests for your requested enhancements. 

- PR #28 - Subquery support
- PR #29 - TYPEOF support

Both of these are still a work in progress, but I wanted to go ahead and share the changes with you now in case you have any feedback. I hope to finish these enhancements within the next week or so.
